ddtrace/profiling/collector/_lock.py

```diff
@@ -106,7 +106,12 @@ def __aenter__(self, *args: Any, **kwargs: Any) -> Any:
 
     def _acquire(self, inner_func: Callable[..., Any], *args: Any, **kwargs: Any) -> Any:
         if not self.capture_sampler.capture():
-            return inner_func(*args, **kwargs)
+            if config.enable_asserts:
+                # Ensure acquired_time is not set to a valid timestamp when acquire is not sampled
+                # (else a bogus release sample is produced)
+                assert self.acquired_time is None, (
+                    f"Expected acquired_time to be None when acquire is not sampled, got {self.acquired_time!r}"
+                )
```

Comment:
```suggestion
                assert self.acquired_time is None, (
                    f"Expected acquired_time to be None when acquire is not sampled, got {self.acquired_time!r}"
                )  # nosec
```

I believe this should suppress the security warning

Reply:
OK - is this comment a blocker for this PR?